### ISSUE TITLE
decryptData() crash

### ISSUE THREAD

**Issue by t** [krzyzanowskim/ObjectivePGP#7]
I'm trying this library. When I call decryptData(), ObjectivePGP crashes. (I might call decryptData with wrong input.)

Environment: XCode 6.1.1 with iOS SDK 8.1 

```
Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[NSConcreteData subdataWithRange:]: range {544, 119296} exceeds data length 1076'
*** First throw call stack:
(
  0   CoreFoundation                      0x000000010ac15f35 __exceptionPreprocess + 165
  1   libobjc.A.dylib                     0x000000010c759bb7 objc_exception_throw + 45
  2   CoreFoundation                      0x000000010ac15e6d +[NSException raise:format:] + 205
  3   Foundation                          0x000000010b059ac7 -[NSData(NSData) subdataWithRange:] + 252
  4   PGPPaste                            0x000000010a8a0a7d +[PGPPacketFactory packetWithData:offset:] + 305
  5   PGPPaste                            0x000000010a899be0 -[ObjectivePGP readPacketsFromData:] + 127
  6   PGPPaste                            0x000000010a89701d -[ObjectivePGP decryptData:passphrase:error:] + 161
  7   PGPPaste                            0x000000010a88895b _TFC8PGPPaste14ViewController11viewDidLoadfS0_FT_T_ + 9243
  8   PGPPaste                            0x000000010a888ab2 _TToFC8PGPPaste14ViewController11viewDidLoadfS0_FT_T_ + 34
  9   UIKit                               0x000000010b5c6a90 -[UIViewController loadViewIfRequired] + 738
  10  UIKit                               0x000000010b5c6c8e -[UIViewController view] + 27
  11  UIKit                               0x000000010b4e5ca9 -[UIWindow addRootViewControllerViewIfPossible] + 58
  12  UIKit                               0x000000010b4e6041 -[UIWindow _setHidden:forced:] + 247
  13  UIKit                               0x000000010b4f272c -[UIWindow makeKeyAndVisible] + 42
  14  UIKit                               0x000000010b49d061 -[UIApplication _callInitializationDelegatesForMainScene:transitionContext:] + 2628
  15  UIKit                               0x000000010b49fd2c -[UIApplication _runWithMainScene:transitionContext:completion:] + 1350
  16  UIKit                               0x000000010b49ebf2 -[UIApplication workspaceDidEndTransaction:] + 179
  17  FrontBoardServices                  0x000000010e2bb2a3 __31-[FBSSerialQueue performAsync:]_block_invoke + 16
  18  CoreFoundation                      0x000000010ab4b53c __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12
  19  CoreFoundation                      0x000000010ab41285 __CFRunLoopDoBlocks + 341
  20  CoreFoundation                      0x000000010ab40a43 __CFRunLoopRun + 851
  21  CoreFoundation                      0x000000010ab40486 CFRunLoopRunSpecific + 470
  22  UIKit                               0x000000010b49e669 -[UIApplication _run] + 413
  23  UIKit                               0x000000010b4a1420 UIApplicationMain + 1282
  24  PGPPaste                            0x000000010a88a11e top_level_code + 78
  25  PGPPaste                            0x000000010a88a15a main + 42
  26  libdyld.dylib                       0x000000010cf33145 start + 1
  27  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by krzyzanowskim** [comment]
I think I know the reason, if so... I can fix it for current version. Will be fixed in upcoming next iteration

**Comment by t** [comment]
:+1:

**Comment by twcctz500000** [comment]
good

**Comment by patrickbgit** [comment]
Was this fixed?

**Comment by alexkunitsa** [comment]
any fix for that?

**Comment by krzyzanowskim** [comment]
no, PR is welcome.

**Comment by mcqueentc** [comment]
Do you still have plans to fix this problem? Or at least can you tell what it might be?

**Comment by alexkunitsa** [comment]
Here' s what I found:
```
- method  encryptData: usingPublicKeys:armored:error: in ObjectivePGP.m
```
As default we have PGPCompressionBZIP2 while creating PGPCompressedPacket. With that option app will crash when encrypting files > 5kb.

If change that option to PGPCompressionZLIB - then no crash. But looks like small files (or text) is ok, but if is bigger - then you will have crash while decrypting. It's possible to fix it, if you make some changes in PGPPacketFactory.m

```
packetWithData: offset:

     if (packetHeaderData.length > 0) {
        finalBodyLength = definedBodyLength;
        if (definedBodyLength == UnknownLength) {
	
	  // todo - it's temp fix for get in that case, because UnknownLength is not always the same
	  if (definedBodyLength > 4200000000) {
             packetHeaderData = nil;
             finalBodyLength = MAX(0.0,packetData.length - offset - packetHeaderData.length);
          }
       }
}
```
Maybe it's completely wrong, but after that there is no crash. 
But we have new issue :) 
If text is long enough (> 5 kb) then decoded string will lost some first sentences. If you will decrypt file - then result file will be broken.

That's all that I found for now. Maybe someone will have other ideas how to make it work.

BTW: if you will decrypt file that was generated with other tool - then result must be ok. I think that there is smth wrong with encryption in ObjectivePGP
